payjoin/Cargo.lock

```diff
@@ -0,0 +1,1258 @@
+# This file is automatically @generated by Cargo.
```

Comment:
Since this is just for the library [I think it can safely be .gitignore-d](https://doc.rust-lang.org/cargo/faq.html#why-do-binaries-have-cargolock-in-version-control-but-not-libraries). Is this necessary this is for expressing MSRV?

Reply:
I agree we should remove and .gitignore `Cargo.lock` for the lib. That's another advantage of removing the workspace is you can keep a `Cargo.lock` there if you want.